### PR TITLE
Clarify Back Matter Resource Resolution Order in Profile Resolution

### LINKED ISSUE #1442: Imprecise description of profile backmatter resolution
### Describe the bug

See https://pages.nist.gov/OSCAL/concepts/processing/profile-resolution/#d2e1377-head, which states

> The output's backmatter MUST be generated by copying in each resource object from the backmatters of the imported catalogs/profiles in top-to-bottom order, then by copying in each resource object from the backmatter of the source profile itself. These objects MUST be processed in the order they are defined in each respective document.

Neither "top-to-bottom order" nor "the order they are defined in each respective document" have sufficient precision. Nor does this describe the order in which the "catalogs/profiles" should be processed (which should likely be the order in which they are cited in the "source profile", namely "document order"), nor the order in which the `back-matter/resource` elements of each "catalogs/profiles" should be processed (likely "document order"), nor the source profile's `back-matter/resource` element processing (likely "document order").

### Who is the bug affecting

Profile resolution aficionados.

### What is affected by this bug

Documentation

### How do we replicate this issue

See https://pages.nist.gov/OSCAL/concepts/processing/profile-resolution/#d2e1377-head.

### Expected behavior (i.e. solution)

The term "document order" should be used.

The order in which "catalogs/profiles" are processed should likely be the "document order" in which they are cited, followed ultimately by the "source profile".

I am uncertain if catalog or profile importation entails transitive catalog and profile importation and resolution, though that seems likely.

### Other comments

A cat attempted to collaborate on this issue but was dissuaded.

Perhaps it is mentioned elsewhere: there is no explicit accommodation of XML comments or processing-instructions, nor of whitespace handling/normalization. For an XML profile resolution a simple accommodation would be to copy all descendant nodes of each the involved `back-matter` elements (in document order).

### PR DESCRIPTION
**Pull request by aj-stein-nist**
# Committer Notes

Closes #1442.

Some order of operations for processing being described in back matter resource resolution and other key operations of the specification are described as 'top-down order' and similar phrases that are approachable but less precise than 'document order' for underlying markup and data languages we use, like XML, where 'document order' has a more precise meaning.

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/usnistgov/OSCAL/blob/main/CONTRIBUTING.md) guidance?
- [x] Have you set "[Allow edits and access to secrets by maintainers
](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/OSCAL/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- ~Have you written new tests for your core changes, as applicable?~ (intended for separate dev work tracked in another task, link later)
- ~Have you included examples of how to use your new feature(s)?~
- ~Have you updated all [OSCAL website](https://pages.nist.gov/OSCAL) and readme documentation affected by the changes you made? Changes to the OSCAL website can be made in the docs/content directory of your branch.~

### CODE REVIEW

**Review by wendellpiez — APPROVED** [review]
So, we actually avoided the term "document order" as a specifically XML concept -- indeed this term does not appear in the XML Recommendation, and is defined formally (tmk) only in the context of XPath/XDM. Since it had no meaning for JSON, we avoided it in favor of spelling it out, i.e. top-down, depth first ordering.

Given this, should we not at least define the term if we are going to use it? Alternatively we could add a note saying "in XML, document order".

https://www.w3.org/TR/1999/REC-xpath-19991116/#data-model
https://www.w3.org/TR/query-datamodel/#document-order

I'm approving because I don't want to block this as an improvement if you consider it to be one.

**Comment by GaryGapinski** [comment]
>…we actually avoided the term "document order…"

If "document order" is considered too formal, perhaps "the order of occurrence within the document" (deficiently) covers XML and JSON (presuming it entails depth first, which seems to be the case). The term _linear_ also comes to mind.

**Comment by wendellpiez** [comment]
Happy to see clearer language crafted. Also not opposed to saying 'document order' if it can be clarified appropriately or perhaps glossed to make sense in an object processing (i.e. JSON or YAML) context.

Note that because order of properties on objects in JSON is not defined, a data caster going from objects back into XML must look at the metaschema to know that parameters come before properties (props), for example (or know ahead of time).

**Comment by aj-stein-nist** [comment]
I made some light changes and also addressed https://github.com/usnistgov/OSCAL/issues/1442#issuecomment-1455247735. @wendellpiez, can I get a quick re-review?

File: src/specifications/profile-resolution/profile-resolution-specml.xml [hunk 1]
```diff
@@ -751,7 +751,7 @@ intermediate:
         </tagging>
         <p><req level="must" id="req-merge-use-first">When the
           &quot;use-first&quot;combination rule is applied, and controls that share a distinctive ID are found, the first control encountered MUST be kept, the rest MUST be discarded.
-          &quot;First&quot; MUST be determined by a top-down, depth-first traversal of the source profile&apos;s import hierarchy.
+          &quot;First&quot; MUST be determined by a top-down, depth-first traversal of the source profile&apos;s in the order they occur within the import hierarchy of the docuent.
```
Comment (aj-stein-nist) [inline src/specifications/profile-resolution/profile-resolution-specml.xml:754]:
```suggestion
          &quot;First&quot; MUST be determined by a top-down, depth-first traversal of the source profile&apos;s in the order they occur within the import hierarchy of the document.
```

File: src/specifications/profile-resolution/profile-resolution-specml.xml [hunk 2]
```diff
@@ -1383,7 +1383,7 @@ control:
       <p>The following requirements MUST be followed with regards to the Metadata section of the output catalog:</p>
       <ul>
         <li>
-          <p><req level="must" id="req-meta-uuid">The output catalog's metadata MUST have a unique top-level UUID (metadata:uuid). This UUID MAY be generated as seen fit by the tool, as long as it is reasonable to assume it is globally unique. It is RECOMMENDED that tools use a Version 4 UUID as specified in <a href="https://datatracker.ietf.org/doc/html/rfc4122">Section 4 of RFC 4122</a>.
+          <p><req level="must" id="req-meta-uuid">The output catalog's metadata MUST have a unique top-level UUID. This UUID MAY be generated as seen fit by the tool, as long as it is reasonable to assume it is globally unique. It is RECOMMENDED that tools use a Version 4 UUID as specified in <a href="https://datatracker.ietf.org/doc/html/rfc4122">Section 4 of RFC 4122</a>.
```
Comment (aj-stein-nist) [inline src/specifications/profile-resolution/profile-resolution-specml.xml:1386]:
```suggestion
          <p><req level="must" id="req-meta-uuid">The output catalog MUST have a unique top-level UUID. This UUID MAY be generated as seen fit by the tool, as long as it is reasonable to assume it is globally unique. It is RECOMMENDED that tools use a Version 4 UUID as specified in <a href="https://datatracker.ietf.org/doc/html/rfc4122">Section 4 of RFC 4122</a>.
```

**Comment by aj-stein-nist** [comment]
This was not completed in time for the remainder for Sprint 64 and higher priorities came up with Sprint 65. I will add the related issue to sprint 66 and reopen the PR or start over accordingly as time permits.

**Pull request closed without merging**